ep/build-velox/src/get_velox.sh

```diff
@@ -189,6 +189,9 @@ function setup_linux {
       ;;
     esac
   elif [[ "$LINUX_DISTRIBUTION" == "openEuler" ]]; then
+    # this is workaround for gcc-12.3.1
+    # https://github.com/facebookincubator/velox/blob/b263d9dd8b8910dc642d8fdb0c0adee4b2a1fb29/CMakeLists.txt#L433
+    sed -i "s|no-unknown-warning-option|no-unknown-warning-option -Wno-restrict|g" ../../src/build_velox.sh
```

Comment:
According to the link below commented in Velox, it seems the gcc community is retargeting the bugs to GCC 12.5. Therefore, I am wondering if we can just submit a Velox PR to include 12.3.x and even 12.4.x in that check of CMakeLists.txt. Assume this issue is not limited to openEuler.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105329

Reply:
@PHILO-HE It seems more complicate than I thought, [105651](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651) mentioned this was fixed in 12.4, not sure if it's the same bug.

Reply:
They look similar. If 12.4 is not confirmed, I think we can just include 12.3 in the check at Velox side.

Reply:
@PHILO-HE @kevinw66 could we land this fix on Gluten side first? My intention is to fix the coming 1.5 release.
I will also submit the patch to Velox, once it's merged we could revert this workaround